aten/src/ATen/native/cpu/BinaryOpsKernel.cpp

```diff
@@ -104,12 +104,22 @@ void logical_xor_kernel(TensorIterator& iter) {
 }
 
 void lt_kernel(TensorIterator& iter) {
-  AT_DISPATCH_ALL_TYPES_AND2(kBool, kBFloat16, iter.input_dtype(), "lt_cpu", [&]() {
-    cpu_kernel(iter,
-     [=](scalar_t a, scalar_t b) -> bool {
-       return a < b;
-     });
-  });
+  if (iter.dtype() == ScalarType::Bool || iter.input_dtype() == ScalarType::Byte) {
+    AT_DISPATCH_ALL_TYPES_AND2(kBool, kBFloat16, iter.input_dtype(), "lt_cpu", [&]() {
```

Comment:
is it intentional that we're adding cpu half support that didn't exist previously?

on master, `torch.tensor(5).half() < 6` would give `RuntimeError: _th_lt not supported on CPUType for Half`

Reply:
https://github.com/pytorch/pytorch/blob/71ec9a0035025bafc02c12f92fff63bdd9735c37/aten/src/ATen/Dispatch.h#L62-L66

This shouldn't add half support for CPU

Reply:
Fixed